src/main/java/org/terasology/taskSystem/TaskManagementSystem.java

```diff
@@ -344,4 +349,56 @@ public boolean assignAdvancedTaskToOreon(Actor oreon, String assignedTaskType) {
 
             return true;
     }
+
+    /**
+     * Receives the HorizontalCollisionEvent and decides whether the Oreon should abandon tasks
+     */
+    @ReceiveEvent(priority = EventPriority.PRIORITY_CRITICAL)
+    public void receiveCollisionEvent(HorizontalCollisionEvent collisionEvent, EntityRef oreon, MinionMoveComponent moveComponent) {
+        if (lastCollisionLocation == null) {
+            lastCollisionLocation = collisionEvent.getLocation();
+        } else {
+            if (lastCollisionLocation.getX() == collisionEvent.getLocation().getX() && lastCollisionLocation.getZ() == collisionEvent.getLocation().getZ()) {
+                logger.info("oreon stuck");
+                moveComponent.target = null;
+                oreon.saveComponent(moveComponent);
+                abandonTask(oreon);
+            } else {
+                //if collision just took place once
+                lastCollisionLocation = null;
+            }
+        }
+    }
+
+    private void abandonTask(EntityRef oreon) {
+        TaskComponent oreonTaskComponent = oreon.getComponent(TaskComponent.class);
+
+        if (!oreonTaskComponent.assignedTaskType.equals(AssignedTaskType.None)) {
+            oreon.getComponent(OreonSpawnComponent.class).parent.getOwner().send(new ChatMessageEvent("Oreon got stuck abandoning task", notificationMessageEntity));
```

Comment:
"Oreon got stuck abandoning task" should be cleaned up to something like "Oreon got stuck.  Abandoning task."  And it would be good if you could identify which oreon.